src/pages/workspace/WorkspaceSidebar.js

```diff
@@ -58,39 +58,64 @@ const defaultProps = {
     allPolicies: null,
 };
 
+/**
+ * @param {Object} navigationState
+ * @returns {String|undefined}
+ */
+function getPolicyIDFromNavigationState(navigationState) {
+    const routes = lodashGet(navigationState, 'routes', []);
+    const routeWithPolicyIDParam = _.find(routes, route => route.params && route.params.policyID);
+    return lodashGet(routeWithPolicyIDParam, ['params', 'policyID']);
+}
+
 const WorkspaceSidebar = ({
-    translate, isSmallScreenWidth, policy, allPolicies, isFocused,
+    translate, isSmallScreenWidth, policy, allPolicies, isFocused, navigation,
 }) => {
+    const policyID = lodashGet(policy, 'id');
     const menuItems = [
         {
             translationKey: 'workspace.common.card',
             icon: ExpensifyCard,
             action: () => {
-                Navigation.navigate(ROUTES.getWorkspaceCardRoute(policy.id));
+                if (policyID) {
+                    Navigation.navigate(ROUTES.getWorkspaceCardRoute(policyID));
+                }
             },
-            isActive: Navigation.isActiveRoute(ROUTES.getWorkspaceCardRoute(policy.id)),
+            isActive: Navigation.isActiveRoute(ROUTES.getWorkspaceCardRoute(policyID)),
         },
         {
             translationKey: 'common.people',
             icon: Users,
             action: () => {
-                Navigation.navigate(ROUTES.getWorkspacePeopleRoute(policy.id));
+                if (policyID) {
+                    Navigation.navigate(ROUTES.getWorkspacePeopleRoute(policyID));
+                }
             },
-            isActive: Navigation.isActiveRoute(ROUTES.getWorkspacePeopleRoute(policy.id)),
+            isActive: Navigation.isActiveRoute(ROUTES.getWorkspacePeopleRoute(policyID)),
         },
     ];
 
-    // After all the policies have loaded, we can know if the given policyID points to a nonexistant workspace
-    // When free plan is out of beta and Permissions.canUseFreePlan() gets removed,
-    // all code involving 'allPolicies' can be removed since policy loading will no longer be delayed on login.
-    if (allPolicies !== null && _.isEmpty(policy)) {
+    /* After all the policies have loaded, check if the given policyID points to a nonexistant workspace.
+     *
+     * Check the route parameter, because we can't tell the difference between a default value for policy not yet loaded
+     * and the scenario where the policy isn't found in Onyx.
+     *
+     * When free plan is out of beta and Permissions.canUseFreePlan() gets removed,
+     * all code involving 'allPolicies' can be removed since policy loading will no longer be delayed on login.
+     */
+    const policyIDFromRoute = getPolicyIDFromNavigationState(navigation.getState());
+    if (allPolicies && !_.any(allPolicies, loadedPolicy => loadedPolicy.id === policyIDFromRoute)) {
```

Comment:
Random thought that probably has an obvious answer but I'm confused about why we need to throw this error at all? 

What invalid workspace situation are we trying to prevent? Could we just remove all this code that throws errors about invalid policies and instead wait for the policy we expect to load to load then show the menu options?

Reply:
Since `allPolicies` is a collection are we sure that this will always include "all the policies"?

e.g. we could set some here:

https://github.com/Expensify/App/blob/64bea5330220a936a16fa049260a0e491baa6c5d/src/libs/actions/Policy.js#L110-L110

or here

https://github.com/Expensify/App/blob/64bea5330220a936a16fa049260a0e491baa6c5d/src/libs/actions/Policy.js#L91-L91

or here

https://github.com/Expensify/App/blob/64bea5330220a936a16fa049260a0e491baa6c5d/src/libs/actions/Policy.js#L232-L232

Reply:
I'm thinking that might be a good option too. I think it's ideal to keep this error, but if just getting rid of it fixes this problem, I'm for it at this point.

Reply:
Yeah I guess I'm just stuck on this part because I agree 💯  this is the solution:

>wait for allPolicies to load, and check if the linked policy (taken from the URL) is included in the list of policies. If not, we assume the policy is invalid.

But I don't see where we are waiting for all the policies to load... 🤔

Reply:
> What invalid workspace situation are we trying to prevent

It's a bit convoluted, but just linking to a workspace that doesn't exist, which can happen if you:


1. Have OldDot open to `admin_policies`
1. In another OldDot tab, delete the workspace
1. In the first OldDot tab, click on the workspace 💥 

> Since allPolicies is a collection are we sure that this will always include "all the policies"?

I'm not sure I follow ... when I say "wait for allPolicies to load", I mean wait for the Onyx data to load for the collection of policy keys. In all of the locations you linked, it seems like Onyx would be updated, and so would the `allPolicies` prop.

Reply:
> Since allPolicies is a collection are we sure that this will always include "all the policies"?

Okay, just caught on. These policies don't always load all at once.